React/src/App.tsx

```diff
@@ -1,17 +1,87 @@
-import React, { useCallback, useState } from 'react';
+import React, {
+  useRef, useMemo, useCallback, RefObject,
+} from 'react';
 import './App.css';
 import 'devextreme/dist/css/dx.material.blue.light.compact.css';
-import Button from 'devextreme-react/button';
+
+import Scheduler, { Resource } from 'devextreme-react/scheduler';
+import { ClickEvent } from 'devextreme/ui/button';
+import { AppointmentTooltipTemplateData } from 'devextreme/ui/scheduler';
+
+import { employees, data, Employee } from './data';
+import DataCell from './DataCell';
+import ResourceCell from './ResourceCell';
+import Tooltip from './Tooltip';
+
+const currentDate = new Date(2021, 5, 2, 11, 30);
+const groups = ['employeeID'];
+const views = ['month'] as ['month'];
+
+function getDisabled(employeeID: number): boolean | undefined {
+  return employees.find((employee: Employee) => employee.id === employeeID)?.disabled;
+}
+
+function getColor(employeeID: number): string | undefined {
+  return employees.find((employee: Employee) => employee.id === employeeID)?.color;
+}
+
+function createAppointmentTooltip(schedulerRef: RefObject<Scheduler>) {
+  return function AppointmentTooltip({ data: props }: { data: AppointmentTooltipTemplateData }): JSX.Element {
+    const scheduler = schedulerRef.current;
+    const onDeleteButtonClick = useCallback((e: ClickEvent): void => {
+      scheduler?.instance.deleteAppointment(props.appointmentData);
+      e.event?.stopPropagation();
+      scheduler?.instance.hideAppointmentTooltip();
+    }, []);
```

Comment:
Let's add ```scheduler``` and ```props.appointmentData``` to ```useCallback``` dependencies
Or add only the ```props.appointmentData``` and use ```schedulerRef``` inside function

Reply:
Resolved